pkg/awsManager/vpcVpnManager.go

```diff
@@ -260,6 +269,9 @@ func DetachAndDeleteNetworkInterface(client clientpkg.Client, vpcID *string, log
 				_, err = client.DeleteNetworkInterface(&ec2.DeleteNetworkInterfaceInput{NetworkInterfaceId: networkInterface.NetworkInterfaceId})
 				if err != nil {
 					logger.Error(err, "Failed to delete Network Interface", *networkInterface.NetworkInterfaceId)
+					localMetrics.ResourceFail(localMetrics.NetworkInterface, client.GetRegion())
+				} else {
+					localMetrics.ResourceSuccess(localMetrics.NetworkInterface, client.GetRegion())
```

Comment:
Don't think we need this else block

Reply:
We need this else block here because there's no continue (or way to `continue`) because this is not in a small enough scoped for block.  If we remove this else statement something that got marked as fail would also get marked as success.

The same idea applies to the other two locations where I had to add else blocks.

Theoretically it could be refactored further in order to utilize a return or continue (because honestly I'm of the camp that "else" is a code smell) but that would require a bit of re-work that's a bit outside the scope of this ticket as well as the lack of test coverage would make me iffy about doing that level of work here.